engine/src/main/java/org/terasology/config/flexible/Setting.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import org.terasology.config.flexible.validators.SettingValueValidator;
+import org.terasology.engine.SimpleUri;
+import org.terasology.utilities.subscribables.GeneralSubscribable;
+
+/**
+ * Represents a setting stored in a {@link FlexibleConfig} identified by an id. Contains a value that can
+ * optionally be validated by a {@link SettingValueValidator<T>} and notifies its subscribers when the value
+ * stored is changed.
+ * @param <T> The type of the value this {@link Setting} contains.
+ */
+public interface Setting<T> extends GeneralSubscribable {
+    /**
+     * Returns the id of this {@link Setting}.
```

Comment:
I just like to annotate all types and code references in documentation with the link tag, because it is syntax highlighted in the documentation comments and looks better when performing a documentation lookup through an IDE. If you don't think it's appropriate I can remove them.